tests/end-to-end/external-solution-integration/internal/scenario/compass_e2e.go

```diff
@@ -0,0 +1,179 @@
+package scenario
+
+import (
+	"crypto/tls"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/vrischmann/envconfig"
+
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/helpers"
+
+	"github.com/kyma-project/kyma/common/resilient"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/internal"
+
+	kubelessClient "github.com/kubeless/kubeless/pkg/client/clientset/versioned"
+	serviceCatalogClient "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+	gatewayClient "github.com/kyma-project/kyma/components/api-controller/pkg/clients/gateway.kyma-project.io/clientset/versioned"
+	appBrokerClient "github.com/kyma-project/kyma/components/application-broker/pkg/client/clientset/versioned"
+	appOperatorClient "github.com/kyma-project/kyma/components/application-operator/pkg/client/clientset/versioned"
+	eventingClient "github.com/kyma-project/kyma/components/event-bus/generated/push/clientset/versioned"
+	serviceBindingUsageClient "github.com/kyma-project/kyma/components/service-binding-usage-controller/pkg/client/clientset/versioned"
+
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/step"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/testkit"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/testsuite"
+	"github.com/spf13/pflag"
+	coreClient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// CompassE2E executes complete external solution integration test scenario
+// using Compass for Application registration and connectivity
+type CompassE2E struct {
+	domain            string
+	testID            string
+	skipSSLVerify     bool
+	applicationTenant string
+	applicationGroup  string
+	lambdaPort        int
+}
+
+// AddFlags adds CLI flags to given FlagSet
+func (s *CompassE2E) AddFlags(set *pflag.FlagSet) {
+	pflag.StringVar(&s.domain, "domain", "kyma.local", "domain")
+	pflag.StringVar(&s.testID, "testID", "compass-e2e-test", "domain")
+	pflag.BoolVar(&s.skipSSLVerify, "skipSSLVerify", false, "Skip verification of service SSL certificates")
+	pflag.StringVar(&s.applicationTenant, "applicationTenant", "", "Application CR Tenant")
+	pflag.StringVar(&s.applicationGroup, "applicationGroup", "", "Application CR Group")
+	pflag.IntVar(&s.lambdaPort, "lambdaPort", 8080, "Lambda port")
+}
+
+// Steps return scenario steps
+func (s *CompassE2E) Steps(config *rest.Config) ([]step.Step, error) {
+	state, err := s.NewState()
+	if err != nil {
+		return nil, err
+	}
+
+	appOperatorClientset := appOperatorClient.NewForConfigOrDie(config)
```

Comment:
I would think about moving all clientSet initialization to other method or struct.
Something like:
```
clients := initClients(config)

and then use it in all steps.
```